### PR TITLE
Object Detection from a video

### PR DESCRIPTION
**Pull request by theiturhs**
# Pull Request for DL-Simplified 💡

## Issue Title : Object Detection from a video

- **Info about the related issue (Aim of the project)** : Identify the objects from an input video with an accuracy more than 90%.
- **Name:** Shruti Shrivastava
- **GitHub ID:** [theiturhs](https://github.com/theiturhs)
- **Email ID:** shrutishrivastava22ss@gmail.com
- **Idenitfy yourself:** `GSSoC'24`

Closes: #11 

### Describe the add-ons or changes you've made 📃

For object detection, three different Convolutional Neural Network architecture based models are implemented to detect objects from video. Initially, they are implemented on images, then on videos to check their detection and with what confidence score it detects and how long it takes to detect objects.

Here are the list of pretrained models that are used:

- MobileNet SSD
- Faster R-CNN
- YOLOv8

## Type of change ☑️

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? ⚙️

So, the detection process was first implemented of images. After verifying the results obtained from images, detection was made on video by taking each frame. The frames where objects are detected are then converted to videos.

## Checklist: ☑️

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by abhisheks008 — APPROVED** [review]
Approved @theiturhs
